File: LSP+ISP/Spoilers/MarsRoverKata/Commands/TurnLeftCommand.cs
namespace MarsRoverKata.Commands
{
    public class TurnLeftCommand : ICommand
    {
        private IMovable Movable { get; set; }

        public TurnLeftCommand(IMovable movable)
        {
            Movable = movable;
        }

        public bool Execute()
        {
            return Movable.TurnLeft();
        }
    }
}

File: DIP/Java_Final/MarsRoverKata/src/main/java/MarsRoverKata/Commands/TurnLeftCommand.java
package MarsRoverKata.Commands;

import MarsRoverKata.IMovable;

public class TurnLeftCommand implements ICommand {
    private IMovable movable;

    public IMovable getMovable() {
        return movable;
    }

    public void setRover(IMovable movable) {
        this.movable = movable;
    }

    public TurnLeftCommand(IMovable movable) {
        this.movable = movable;
    }

    public boolean execute() {
        return movable.turnLeft();
    }
}

